src/core/jsonld.js

```diff
@@ -0,0 +1,91 @@
+// Module core/jsonld
+// Create script element with JSON-LD for SEO document description
+export const name = "core/jsonld";
+
+export function run(conf, doc, cb) {
+  if (!conf.doJsonLd) {
+    return cb();
+  }
+  // Content for JSON
+  const types = ["TechArticle"];
+  if (conf.rdfStatus) types.push(conf.rdfStatus);
+
+  const jsonld = {
+    "@context": ["http://schema.org", {
+      "@vocab": "http://schema.org/",
+      "@language": "en",
+      "w3p": "http://www.w3.org/2001/02pd/rec54#",
+      "foaf": "http://xmlns.com/foaf/0.1/",
+      "datePublished": {"@type": "xsd:date"},
+      "inLanguage": {"@language": null},
+      "isBasedOn": {"@type": "@id"},
+      "license": {"@type": "@id"}
+    }],
+    id: conf.canonicalURI || conf.thisVersion,
+    type: types,
+    name: conf.title,
+    inLanguage: doc.documentElement.getAttribute("lang") || "en",
+    license: conf.licenseInfo.url,
+    datePublished: conf.dashDate,
+    copyrightHolder: {
+      name: "World Wide Web Consortium",
+      url: "https://www.w3.org/"
+    },
+    discussionUrl: conf.issueBase
+  };
+
+  if (conf.subtitle) jsonld.alternativeHeadline = conf.subtitle;
+  if (conf.prevVersion) jsonld.isBasedOn = conf.prevVersion;
+
+  // description from abstract
+  const $abs = doc.getElementById("abstract");
+  if ($abs && $abs.textContent.length > 0) {
+    jsonld.description = $abs.textContent;
+  }
+
+  // Editors
+  if (conf.editors) {
+    jsonld.editor = conf.editors.map(addPerson);
+  }
+  if (conf.authors) {
+    jsonld.contributor = conf.authors.map(addPerson);
+  }
+
+  // normative and informative references
+  const refs = [...conf.normativeReferences, ...conf.informativeReferences];
+  jsonld.citation = refs.map(ref => addRef(conf, ref));
+
+  var $jsonld = hyperHTML`<script type="application/ld+json"></script>`;
+  $jsonld.appendChild(doc.createTextNode(JSON.stringify(jsonld)));
+  doc.head.appendChild($jsonld);
+
+  cb();
+}
+
+// Turn editors and authors into a list of JSON-LD relationships
+function addPerson(person) {
```

Comment:
This one can just become:
```JS
function addPerson({ name, url, mailto, company, companyURL }) {
  const ed = {
    type: "Person",
    name,
    url,
    "foaf:mbox": mailto,
    worksFor: {
      name: company,
      url: companyURL,
    },
  };
  return ed;
}
```

When `company` and `companyURL` are missing, `worksFor` will be an empty object when JSON.stringify() is called, which should be fine.

Reply:
`worksFor` is an empty object, but that's not reasonable JSON-LD, IMHO. It implies that the person works for something, which is not what's asserted.